### PR TITLE
Add page for videos in web entitys

### PR DESCRIPTION
**Pull request by jug007**

### CODE REVIEW

**Comment by jug007** [comment]
Opening this pull request now as I have not been able to find more information on embedding videos properly into Vircadia.
Hoping on https://github.com/brandonocasey/videojs-group-play/issues/2 and https://github.com/makitsune/hifi-things/issues/3 for adding one of those later.

**Comment by jug007** [comment]
Oh and also, as I cannot currently build the docs locally, I would appreciate it if someone could build this and report any issues.

File: docs/source/create/entities/web-entity.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
+##########################
+Play media in a Web Entity
+##########################
+
+Web entities can play a number of different media formats through embedded players.
+
+.. contents:: On This Page
+    :depth: 1
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:8]: The "On This Page" should exclude the page's title and include the section titles.

File: docs/source/create/entities/web-entity.rst [hunk 2]
```diff
@@ -0,0 +1,130 @@
+##########################
+Play media in a Web Entity
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:2]:
```suggestion
Play Media in a Web Entity
```

File: docs/source/create/entities/web-entity.rst [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+##########################
+
+Web entities can play a number of different media formats through embedded players.
+
+.. contents:: On This Page
+    :depth: 1
+
+-----------------------
+Supported media formats
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:11]:
```suggestion
Supported Media Formats
```

File: docs/source/create/entities/web-entity.rst [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+-----------------------
+
+Currently exclusively open source formats are supported:
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:14]:
```suggestion
Currently, exclusively open source formats are supported:
```

File: docs/source/create/entities/web-entity.rst [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
++------------------------+------------------------------------------------------------------+--------------------------+
+| Container format       | Supported codecs                                                 | File extension(s)        |
++========================+==================================================================+==========================+
+| WebM                   | Video: VP8, VP9  Audio: Vorbis, Opus                             | .webm                    |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:19]: Put the "Audio:" section on a new line.
Ditto for similar table content.

File: docs/source/create/entities/web-entity.rst [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+| Matroska               | Video: Theora, VP8, VP9  Audio: Vorbis, Opus, FLAC, MP3          | .mkv, .mk3d, .mka, .mks  |
++------------------------+------------------------------------------------------------------+--------------------------+
+| Ogg                    | Video: Theora, VP8  Audio: Vorbis, Opus, FLAC                    | .ogg, .oga, .ogv         |
++------------------------+------------------------------------------------------------------+--------------------------+
+
++------------------------+------------------------------------------------------------------+--------------------------+
+| Video codec            | Comments                                                         | Supported containers     |
++========================+==================================================================+==========================+
+| VP9 (recommended)      | Good quality, slow encoding                                      | WebM, Matroska           |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:29]:
```suggestion
| VP9 (recommended)      | Good quality, slow encoding.                                     | WebM, Matroska           |
```

File: docs/source/create/entities/web-entity.rst [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+| VP8                    | Recommended if your APU has hardware accelerated encoding for    | WebM, Matroska           |
+|                        | VP8, but not VP9. Slightly faster than VP9.                      |                          |
+|                        | Slightly lower quality to compression ratio than VP9.            |                          |
++------------------------+------------------------------------------------------------------+--------------------------+
+| Theora                 | Low quality, low compression, outdated                           | Matroska                 |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:35]:
```suggestion
| Theora                 | Low quality, low compression, outdated.                          | Matroska                 |
```

File: docs/source/create/entities/web-entity.rst [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+
++------------------------+------------------------------------------------------------------+--------------------------+
+| Audio codec            | Comments                                                         | Supported containers     |
++========================+==================================================================+==========================+
+| Opus                   | Recommended, works well with high and low bitrate                | WebM, Matroska, Ogg      |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:41]:
```suggestion
| Opus                   | Recommended, works well with high and low bitrate.               | WebM, Matroska, Ogg      |
```

File: docs/source/create/entities/web-entity.rst [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+| Vorbis                 | Lower quality to compression ratio than Opus                     | WebM, Matroska, Ogg      |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:43]:
```suggestion
| Vorbis                 | Lower quality to compression ratio than Opus.                    | WebM, Matroska, Ogg      |
```

File: docs/source/create/entities/web-entity.rst [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+| FLAC                   | Lossless, bigger files                                           | Matroska, Ogg            |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:45]:
```suggestion
| FLAC                   | Lossless, bigger files.                                          | Matroska, Ogg            |
```

File: docs/source/create/entities/web-entity.rst [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+| MP3                    | Lower quality to compression ratio than Opus and Vorbis          | Matroska                 |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:47]:
```suggestion
| MP3                    | Lower quality to compression ratio than Opus and Vorbis.         | Matroska                 |
```

File: docs/source/create/entities/web-entity.rst [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
++------------------------+------------------------------------------------------------------+--------------------------+
+
+--------------------------
+Reencode media with ffmpeg
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:51]:
```suggestion
Re-encode Media with FFmpeg
```

File: docs/source/create/entities/web-entity.rst [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+--------------------------
+
+Here are some example commands for reencoding video files using ffmpeg:
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:54]:
```suggestion
Here are some example commands for reencoding video files using FFmpeg:
```

File: docs/source/create/entities/web-entity.rst [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+
+    echo "" | ffmpeg -i "$ITEM" -c:v libvpx-vp9 -b:v 0 -crf 5 -vf scale=-2:600 -cpu-used 5 -row-mt 1 -c:a libopus -b:a 96K "$OUTPUT"
+
+    # All files with the listed extensions will be reencoded.
+    done< <(find "$SOURCE" \( -iname '*.mp4' -or -iname '*.avi'  -or -iname '*.mkv' -or -iname '*.mts' \) -print0) |
+
+.. note:: Do not forget to set the script as executable on Linux using something like `chmod +x vircadia_reencode.sh`
+
+------------
+Video player
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:115]:
```suggestion
Video Player
```

File: docs/source/create/entities/web-entity.rst [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+You can embed videos using the HTML5 video tag. This example code can be put into an HTML file on a web server and be used in a web entity::
+
+    <video controls="controls" src="EXAMPLEDIRECTORY/EXAMPLEFILE.webm">
+        Your browser does not support the HTML5 Video element.
+    </video>
+
+
+**See Also**
+
++ :doc:`Create new Entity's <create-entities>`
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:127]:
```suggestion
+ :doc:`Create new Entities <create-entities>`
```

File: docs/source/create/entities/web-entity.rst [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+| MP3                    | Lower quality to compression ratio than Opus and Vorbis          | Matroska                 |
++------------------------+------------------------------------------------------------------+--------------------------+
+
+--------------------------
+Reencode media with ffmpeg
+--------------------------
+
+Here are some example commands for reencoding video files using ffmpeg:
+
++------------------------+------------------------------------------------------------------+--------------------------+
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:56]: This table isn't displaying in the compiled page output. It just looks like the table formatting doesn't have all the vertical bars etc. lined up.

File: docs/source/create/entities/web-entity.rst [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+Additional info can be found with this command ``ffmpeg -h``
+    
+Alternatively you can explore these links:
+
+* https://trac.ffmpeg.org/wiki
+* https://ffmpeg.org
+* https://wiki.archlinux.org/index.php/Hardware_video_acceleration#VA-API_drivers
+
+------------------------
+Batch processing
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:79]:
```suggestion
Batch Processing
```

File: docs/source/create/entities/web-entity.rst [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+------------------------
+
+Here is an example shell script to batch reencode the contents of a folder on Linux using ffmpeg:
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:82]:
```suggestion
Here is an example shell script to batch reencode the contents of a folder on Linux using FFmpeg.
```

File: docs/source/create/entities/web-entity.rst [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+
+You can name it ``vircadia_reencode.sh``
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:84]:
```suggestion
You can name it: ``vircadia_reencode.sh``
```

File: docs/source/create/entities/web-entity.rst [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+
+With this code in it::
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:86]:
```suggestion
With this code in it:
```

File: docs/source/create/entities/web-entity.rst [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+    # All files with the listed extensions will be reencoded.
+    done< <(find "$SOURCE" \( -iname '*.mp4' -or -iname '*.avi'  -or -iname '*.mkv' -or -iname '*.mts' \) -print0) |
+
+.. note:: Do not forget to set the script as executable on Linux using something like `chmod +x vircadia_reencode.sh`
+
+------------
+Video player
+------------
+
+You can embed videos using the HTML5 video tag. This example code can be put into an HTML file on a web server and be used in a web entity::
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:118]:
```suggestion
You can embed videos using the HTML5 video tag. This example code can be put into an HTML file on a web server and be used in a Web entity::
```

File: docs/source/create/entities/web-entity.rst [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+
+    <video controls="controls" src="EXAMPLEDIRECTORY/EXAMPLEFILE.webm">
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity.rst:120]: May want to apply the same code formatting fix as used for the above.

File: docs/source/create/entities/web-entity.rst (hunk 20 above)
Comment (jug007) [reply docs/source/create/entities/web-entity.rst]: This is supposed to be that way. It is a minimalized way to start a literal block. 
I will change it to the partially minimized form anyways, to make it more obvious.
https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks

File: docs/source/create/entities/web-entity.rst (hunk 5 above)
Comment (jug007) [reply docs/source/create/entities/web-entity.rst]: I don't actually see any way to do this. I just added some more spacing between to make it easier to read and potentially do a new line on its own.

File: docs/source/create/entities/web-entity.rst (hunk 1 above)
Comment (jug007) [reply docs/source/create/entities/web-entity.rst]: I don't actually get what you mean with this

Comment (jug007) [reply docs/source/create/entities/web-entity.rst]: Nevermind, working on it

Comment (jug007) [reply docs/source/create/entities/web-entity.rst]: I cannot actually find a way to do this. I changed it to show the section titles. All the other pages in rst format also show the page title, so I would just leave it like this.

File: docs/source/create/entities/web-entity-media.rst [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+.. code-block:: html
+
+    <video controls="controls" src="EXAMPLEDIRECTORY/EXAMPLEFILE.webm">
+        Your browser does not support the HTML5 Video element.
+    </video>
+
+
+**See Also**
+
++ :doc:`Create new Entites <create-entities>`
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity-media.rst:158]:
```suggestion
+ :doc:`Create New Entites <create-entities>`
```

File: docs/source/create/entities/web-entity-media.rst [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+|                        |                                                                  | resolution while keeping |
+|                        |                                                                  | the aspect ratio.        |
+|                        |                                                                  | "-crf 5" is the video    |
+|                        |                                                                  | quality from 0 to 63,    |
+|                        |                                                                  | lower being better. For  |
+|                        |                                                                  | the constant quality to  |
+|                        |                                                                  | work the bitrate has to  |
+|                        |                                                                  | be set to "0" via        |
+|                        |                                                                  | "-b:v 0"                 |
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/web-entity-media.rst:70]:
```suggestion
|                        |                                                                  | "-b:v 0".                |
```

File: docs/source/create/entities/create-entities.md [hunk 25]
```diff
@@ -39,8 +39,9 @@ You can choose from the following entity types:
 + [Add a Material Entity](material-entity.html)
 + [Change How Entities Look](entity-appearance.html)
 + [Import Your 3D Model](../3d-models/import-model.html)
++ [Play media in a Web Entity](web-entity.html)
```
Comment (ctrlaltdavid) [inline docs/source/create/entities/create-entities.md:42]:
```suggestion
+ [Play Media in a Web Entity](web-entity-media.html)
```

**Pull request merged**